version: '3.8'

services:
  web:
    build:
      context: .
      dockerfile: docker/Dockerfile
    restart: always
    container_name: mnist-digit-recognizer-web-1
    ports:
      - "8501:8501"
    depends_on:
      - db
    environment:
      - DB_HOST=db
      - DB_PORT=${DB_PORT}
      - DB_NAME=${DB_NAME}
      - DB_USER=${DB_USER}
      - DB_PASSWORD=${DB_PASSWORD}
    volumes:
      - ./saved_models:/app/saved_models
    networks:
      - mnist-network

  db:
    image: postgres:13
    restart: always
    container_name: mnist-digit-recognizer-db-1
    environment:
      - POSTGRES_DB=${DB_NAME}
      - POSTGRES_USER=${DB_USER}
      - POSTGRES_PASSWORD=${DB_PASSWORD}
    volumes:
      - postgres_data:/var/lib/postgresql/data
      - ./database/init.sql:/docker-entrypoint-initdb.d/init.sql
    ports:
      - "${DB_PORT}:5432"
    networks:
      - mnist-network
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${DB_USER}"]
      interval: 5s
      timeout: 5s
      retries: 5

networks:
  mnist-network:
    driver: bridge

volumes:
  postgres_data: 